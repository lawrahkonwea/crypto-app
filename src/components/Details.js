import React from 'react';
import { useLocation, Link } from 'react-router-dom';
import { BsFillArrowLeftCircleFill } from 'react-icons/bs';
/* eslint-disable-next-line */
const Details = () => {
  const location = useLocation();
  if (location.state) {
    const {
      // eslint-disable-next-line
      name, image, description, country, year_established, trust_score_rank, trade_volume_24h_btc,
    } = location.state;

    return (
      <div className="details">
        <button type="button">
          <Link className="back-arrow" to="/"><BsFillArrowLeftCircleFill /></Link>
        </button>
        <div className="header-detail">
          <img className="img-detail" src={image} alt={name} />
        </div>
        <table>
          <caption>{name}</caption>
          <tbody>
            <tr>
              <p>country origin:</p>
              <p className="desc-color">
                {
                  country ? (
                    <p className="desc-color">

                      <span>{country}</span>
                    </p>
                  ) : ''
                }
              </p>
            </tr>
            <tr>
              <p>
                {
                  description ? (
                    <p>
                      Description:
                      <span>{description}</span>
                    </p>
                  ) : 'No Description'
                }
              </p>
            </tr>

            <tr>
              <p>Ranked at</p>
              <span>
                #
                {/* eslint-disable-next-line */}
                {trust_score_rank}
              </span>
            </tr>
            <tr>
              <p>Trade volume 24h btc:</p>
              {/* eslint-disable-next-line */}
              <span>{trade_volume_24h_btc}</span>
            </tr>
            <tr>
              <p>Year Established:</p>
              <p>

                {
                  // eslint-disable-next-line
                  year_established ? (
                    <p>
                      {/* eslint-disable-next-line */}
                      <span>{year_established}</span>
                    </p>
                  ) : ''
                }
              </p>
            </tr>
          </tbody>
        </table>
      </div>

    );
  }
};

export default Details;
