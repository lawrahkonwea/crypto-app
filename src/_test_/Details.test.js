import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import { waitFor, render, screen } from '@testing-library/react';
import store from '../redux/configureStore';
import Details from '../components/Details';

test('To test for Detailspage render', () => {
  render(
    <MemoryRouter>
      <Provider store={store}>
        <Details />
      </Provider>
    </MemoryRouter>,
  );
  waitFor(() => expect(screen.getAllByText(/Description:/)).toBeInTheDocument());
});

test('expect Details component to match snapshot', () => {
  const view = renderer.create(
    <MemoryRouter>
      <Provider store={store}>
        <Details />
      </Provider>
    </MemoryRouter>,
  ).toJSON();
  expect(view).toMatchSnapshot();
});
