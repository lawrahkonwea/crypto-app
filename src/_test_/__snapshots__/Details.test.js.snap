// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect Details component to match snapshot 1`] = `null`;
